llms/mlx_lm/generate.py

```diff
@@ -66,6 +69,13 @@ def setup_arg_parser():
         action="store_true",
         help="Use the default chat template",
     )
+    parser.add_argument(
+        "--prompt-only",
```

Comment:
The name of this argument is a bit confusing to me. I think it might be preferrable to make it a `--verbose` flag which defaults to `True`. If `False` then only the generation is printed with no streaming.

Reply:
@awni  Ok.  It was the fact that we were calling generate with verbose hardcoded to True that made me consider another name.  If we want to keep the defaults as True, what about having --verbose/--no-verbose options . I.e, (Python 3.9+):

parser.add_argument('--verbose', action=argparse.BooleanOptionalAction, default=True)

Reply:
That's really nice, I didn't know about it. But we'd like to support Python 3.8 so it won't work.

You could do something like this:

```
    def str2bool(string):
        return string != "False"

    parser.add_argument(
        "--verbose",
        type=str2bool,
        default=True,
        help="Log verbose output when 'True' or only print the response when 'False'",
    )

Reply:
@awni I have made this change